versions/development/SPEC.md

```diff
@@ -1049,6 +1057,43 @@ output {
 }
 ```
 
+#### Directory Outputs
+
+Directories can be declared as outputs just like files. For example:
+```
+output {
+  Directory d = "created/directory/"
+}
+```
+
+Notes:
+
+ - When a directory is output, the contents (including subdirectories) are considered part of the output but the path is not. 
+ - When a directory is output, its contents *do not overwrite* any pre-existing directory content. For example:
+
+```wdl
+workflow directory_manipulation {
+  # Make a directory:
+  call make_directory
+  Directory d = make_directory.made_directory
+  
+  # List the contents before the "update_directory" task:
+  call list_contents as list_contents_before { input: d = d }
+  String d_contents_before = list_contents_before.list
+
+  # An update task:
+  call update_directory { input: d = d }
+  
+  # List the contents again:
+  call list_contents as list_contents_after { input: d = d }
+  String d1_contents_after = list_contents_after.list
+
+  # No matter what update directory does, the following is *always* true:
+  Boolean match = d1_contents_before == d1_contents_after
```

Comment:
Directory equalities represent a bit of an issue and I am not 100% certain they really should be implemented. They really will depend on the engine Implementation. IE if the engine is not packaging the directories into any format like a `tar` ball or `zip` folder, do we expect it to recursively dive into the directory and compute md5 sums on all of the content?

Reply:
This is a string equality though.

Reply:
@EvanTheB Ah, I see., I interpreted this as Directory Equality

Reply:
hmm, this example is turning out to be problematic in a number of ways (see my first comment), so I might just remove it.

The intention was to have this demonstrate that the directories were immutable - meaning that they are still identical even after a task "changes" one - but I agree that that's going to be really hard to actually implement in a reasonably efficient way. 

So maybe for this first implementation we disallow equality checks and maybe revisit this if there's a huge clamor for it?

Reply:
@cjllanwarne it makes sense to me to disallow equality checks for directories. To me, it is not that informative to know that some content in the directory changed (this might be different for other people), since that change does not really reflect a meaningful state necessarily.

Reply:
Regarding equality, it looks like the spec doesn't say what `File == File` is comparing but the intention was certainly path equality and not content equality. Thus the same should hold for Directory types.

Whether or not these operators should exist at all is another question, but File and Directory should at least be managed the same